internal/staticanalysis/obfuscation/parsing/babel-parser.js

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env node
+/* eslint no-var: 0 */
+
+const parser = require("@babel/parser");
+const fs = require("fs");
+
+function locationString(node) {
+    return (node.loc != null) ? `[${node.loc.start.line},${node.loc.start.column}]` : "[]"
+}
+
+const parseOutputLines = []
+
+function logJSON(type, subtype, name, pos, array, extra = null) {
+    const extraValue = (extra !== null) ? `${JSON.stringify(extra)}` : "{}"
+    const arrayValue = (array !== null) ? array : false
+    const json = `{"type":"${type}","subtype":"${subtype}","data":${JSON.stringify(name)},"pos":${pos},` +
+        `"array":${arrayValue}, "extra":${extraValue}}`
+    parseOutputLines.push(json)
+}
+function logIdentifierJSON(subtype, name, pos, extra = null) {
+    logJSON("Identifier", subtype, name, pos, null, extra)
+}
+
+function logLiteralJSON(subtype, value, pos, inArray, extra = null) {
+    logJSON("Literal", subtype, value, pos, inArray, extra)
+}
+
+function logParameters(node) {
+    const n = node
+    for (let i = 0; i < n.params.length; i++) {
```

Comment:
nit: you can probably do:

```javascript

for (let p of n.params) {
}
```

instead.

Reply:
If I do this it breaks the tests... :/